corehq/apps/reports/views.py

```diff
@@ -162,6 +167,39 @@
 require_can_view_all_reports = require_permission(HqPermissions.view_reports)
 
 
+def _can_view_form_attachment():
+    def decorator(view_func):
+        @wraps(view_func)
+        def _inner(request, domain, *args, **kwargs):
+            if VIEW_FORM_ATTACHMENT.enabled(domain):
+                user = request.couch_user
+                user_allowed = False
+
+                domain_membership = user.get_domain_membership(domain, allow_enterprise=True)
+
+                if domain_membership:
+                    user_allowed = True
+                elif user.is_global_admin() and (domain is None or not domain_restricts_superusers(domain)):
+                    user_allowed = True
+
+                if not user_allowed:
+                    return HttpResponseForbidden()
+
```

Comment:
Now that an auth decorator is always applied, you can drop all this stuff
```suggestion
```

Reply:
O, I see now that it's checked with `api_auth`. Cool! Less code!